tools/aie2xclbin/XCLBinGen.cpp

```diff
@@ -197,55 +198,113 @@ static LogicalResult generateCoreElfFiles(ModuleOp moduleOp,
-        return coreOp.emitOpError("failed to generate ld script for core (")
-               << col << "," << row << ")";
+      std::vector<std::string> extractedIncludes;
+      {
+        auto bcfFileIn = openInputFile(bcfPath, &errorMessage);
+        if (!bcfFileIn) {
+          moduleOp.emitOpError(errorMessage);
+        }
+
+        std::string bcfFile = std::string(bcfFileIn->getBuffer());
+        std::regex r("^_include _file (.*)", std::regex::multiline);
+        auto begin = std::sregex_iterator(bcfFile.begin(), bcfFile.end(), r);
+        auto end = std::sregex_iterator();
+        for (std::sregex_iterator i = begin; i != end; ++i) {
+          extractedIncludes.push_back(i->str(1));
+        }
       }
-      ldscript_output->keep();
-    }
 
-    // We are running a clang command for now, but really this is an lld
-    // command.
-    SmallString<64> elfFile(TK.TempDir);
-    sys::path::append(elfFile, elfFileName);
-    {
-      std::string targetLower = StringRef(TK.TargetArch).lower();
-      SmallVector<std::string, 10> flags;
-      flags.push_back("-O2");
-      std::string targetFlag = "--target=" + targetLower + "-none-elf";
-      flags.push_back(targetFlag);
-      flags.emplace_back(objFile);
-      SmallString<64> meBasicPath(TK.InstallDir);
-      sys::path::append(meBasicPath, "aie_runtime_lib", TK.TargetArch,
-                        "me_basic.o");
-      flags.emplace_back(meBasicPath);
-      SmallString<64> libcPath(TK.PeanoDir);
-      sys::path::append(libcPath, "lib", targetLower + "-none-unknown-elf",
-                        "libc.a");
-      flags.emplace_back(libcPath);
-      flags.push_back("-Wl,--gc-sections");
-      std::string ldScriptFlag = "-Wl,-T," + std::string(ldscript_path);
-      flags.push_back(ldScriptFlag);
-      flags.push_back("-o");
-      flags.emplace_back(elfFile);
-      SmallString<64> clangBin(TK.PeanoDir);
-      sys::path::append(clangBin, "bin", "clang");
-      if (runTool(clangBin, flags, TK.Verbose) != 0) {
-        return coreOp.emitOpError("failed to link elf file for core(")
-               << col << "," << row << ")";
+      SmallString<64> chessWrapperBin(TK.InstallDir);
+      sys::path::append(chessWrapperBin, "bin", "xchesscc_wrapper");
+      SmallString<64> chessworkDir(TK.TempDir);
+      sys::path::append(chessworkDir, "chesswork");
+
+      SmallVector<std::string> flags{StringRef(TK.TargetArch).lower(),
+                                     "+w",
+                                     std::string(chessworkDir),
+                                     "-d",
+                                     "+l",
+                                     std::string(bcfPath),
+                                     "-o",
+                                     std::string(elfFile),
+                                     "-f",
+                                     std::string(objFile)};
+      for (const auto &inc : extractedIncludes) {
+        flags.push_back(inc);
+      }
```

Comment:
nit: can you remove braces for all single line for/ifs (ie follow upstream conventions...)

Reply:
@erieaton-amd thanks :)